add unpacker code for OT

sarafiorendi opened the pull request:
#### PR description:

Main code unpacking OT FEDRawData is EventFilter/Phase2TrackerRawToDigi/plugins/RawToClusterProducer.cc. Two classes to store tracker header and channel offsets are also included.

Adding also an EDAnalyzer to create ntuples containing cluster properties, in order to validate re-digi clusters with original ones.

There are still many commented out couts that will be deleted once we move from draft to official PR.

#### PR validation:

Re-digi vs original clusters are available [here](https://fiorendi.web.cern.ch/ot_unpacker/new_unpacker/validation_allDTCs/), there are still some differences from clusters with large radius ( > 80 cm), which are being checked (NB: comparison is done for clusters with size <=8).

tomalin commented on EventFilter/Phase2TrackerRawToDigi/plugins/Phase2TrackerDumpDigi.cc:
```diff
@@ -0,0 +1,209 @@
+#include "DataFormats/Common/interface/Handle.h"
```
All classes should have a comment at the top explaining what they are.
Also, doesn't this dump clusters? So why the file name?

sarafiorendi replied:
renamed, comment added

tomalin commented on EventFilter/Phase2TrackerRawToDigi/test/Phase2TrackerDumpDigi_cfg.py:
```diff
@@ -0,0 +1,77 @@
+import FWCore.ParameterSet.Config as cms
```
All jobs should have a comment at the top explaining what they do. The file name suggests that this one dumps the digis. But does it really dump clusters?

sarafiorendi replied:
renamed, comment added

tomalin commented on EventFilter/Phase2TrackerRawToDigi/test/SLinkProducerAndUnpacker_cfg.py:
```diff
@@ -0,0 +1,137 @@
+#!/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_0_0_pre2/bin/el9_amd64_gcc12/cmsRun
+## Stub Occupancy Script
```
Longer comment needed to explain what this job does.

sarafiorendi replied:
added comment

tomalin commented on EventFilter/Phase2TrackerRawToDigi/interface/Phase2TrackerSpecifications.h:
```diff
@@ -26,7 +26,7 @@ namespace Phase2TrackerSpecifications
     static const int CIC_Z_BOUNDARY_STRIPS = 0;
 
-    static const int MAX_DTC_ID = 216;
+    static const int MAX_DTC_ID = 217;
```
This is wierd. Since there are 216 DTCs in the Tracker, why does the ID run from 1 to 217?

sarafiorendi replied:
we could choose to loop from 0 to 216 or from 1 to 217 to run over all the DTCs, as the various loops are currently from MIN_DTC_ID to < MAX_DTC_ID. Or change "<" to "<=" in the loops. I can change according to the general preference.

tomalin replied:
Why can't we determine the number of DTCs from the Tracker cabling map?
I would avoid 217. So prefer "<=" if that allows you to use 216.

sarafiorendi replied:
changed

tomalin commented on EventFilter/Phase2TrackerRawToDigi/plugins/RawToClusterProducer.cc:
```diff
@@ -0,0 +1,465 @@
+#include "FWCore/Framework/interface/Event.h"
```
All .h files (or cc files if there is no .h file) should have a comment near the to explaining what they do.

sarafiorendi replied:
added a comment

tomalin commented:
Hi @sarafiorendi , aside from my very minor comments above, this looks good to me. I suggest you remove the "draft" tag, unless it's not ready to be merged for some reason? @amastron23 do you have any comments on this code?

amastron23 commented:
> Hi @sarafiorendi , aside from my very minor comments above, this looks good to me. I suggest you remove the "draft" tag, unless it's not ready to be merged for some reason? @amastron23 do you have any comments on this code?

No, looks good to me too. Just approved.

sarafiorendi commented:
@tomalin thanks, I included your suggestions.
I'm fine to merge this, as I've seen that the remaining differences in 2S clusters are due to some clusters not entering the packing stage, and therefore not being unpacked. I'd suggest we try to address this with @amastron23 in a separate development.

tomalin commented:
Hi @sarafiorendi , two of the comments have not yet been answered, so I can't merge. Can you please look at (1) https://github.com/P2-Tracker-BES-SW/cmssw/pull/5#discussion_r1878313212 (2) https://github.com/P2-Tracker-BES-SW/cmssw/pull/5#discussion_r1878317958  .

sarafiorendi commented:
hi Ian,
I added already the comment and moved back to max DTC 216 in my two commits
https://github.com/P2-Tracker-BES-SW/cmssw/pull/5/commits/ca0f09dd0de5f6832584645b208b6dbd24d93527 
https://github.com/P2-Tracker-BES-SW/cmssw/pull/5/commits/de13203a4b1f2b4a2fb4e24d4feb77b32489c316

tomalin reviewed (approved):
Approved
